client/upgrade/cli/query.go

```diff
@@ -90,29 +97,46 @@ func GetCmdQuerySignals(storeName string, cdc *codec.Codec) *cobra.Command {
 				return err
 			}
 
-			var validatorAddrs []string
+			validatorConsAddrs := make(map[string]bool)
 			res, err := cliCtx.QuerySubspace(upgrade.GetSignalPrefixKey(upgradeConfig.Protocol.Version), storeName)
 			if err != nil {
 				return err
 			}
 
 			for _, kv := range res {
-				validatorAddrs = append(validatorAddrs, upgrade.GetAddressFromSignalKey(kv.Key))
+				validatorConsAddrs[upgrade.GetAddressFromSignalKey(kv.Key)] = true
 			}
 
-			if len(validatorAddrs) == 0 {
+			if len(validatorConsAddrs) == 0 {
 				fmt.Println("No validators have started the new version.")
```

Comment:
"No validator has started the new version."
Or
"Collect no validator upgrade signal"

Reply:
Ok